### DISCUSSION TITLE
Retrieving Media (Direct Messages)

### DISCUSSION THREAD

**Discussion by thesrs02** [plhery/node-twitter-api-v2#219] [Q&A] [answered]
I couldn't find any endpoint wrapper for [Retrieving Media (Direct Messages) ](https://developer.twitter.com/en/docs/twitter-api/v1/direct-messages/message-attachments/guides/retrieving-media)

Is there a way to get the media using this library or do we need to build this part on our own?

Thanks in advance!

**Comment by alkihis** [comment, marked as answer]
Yep, no wrapper exists for `ton.twitter.com`, you're right!

You can download them using the lib, with a little trick it should work.
This isn't optimal, but I welcome it as a feature request!

```ts
// login with OAuth1.0a
const client = new TwitterApi({ appKey: '', appSecret: '', accessToken: '', accessSecret: '' })

// Get your dms
// ...

// Download a media protected by OAuth
const imgRaw = await client.get<string>(dm.media_url_https, undefined, { prefix: '' })
const imgAsBuffer = Buffer.from(imgRaw)

// use/save imgAsBuffer
```

Tell me if it works :)
